test/Hamiltonians.jl

```diff
@@ -1621,7 +1621,7 @@ end
     h2 = ExtendedHubbardReal1D(OccupationNumberFS(3, 0, 1); u=6 + 3im, t=0)
     # diagonal and non-Hermitian
     @test LOStructure(h) isa IsDiagonal
-    @test_throws ArgumentError adjoint(h2)
+    @test LOStructure(h2) isa AdjointKnown
```

Comment:
It would be good to test that the `adjoint`s of the non-hermitian 1D Hubbard models work as expected. At least I would add them to the list of "Hamiltonian interface tests" where it is tested that the `adjoint` doesn't error. In addition it would be good to check that the diagonal are what they should be.

Reply:
It's always good to check the Coverage report as well. You can see that `LOStructure` and `adjoint` of `HubbardReal1DEP` is missing tests.